File: javascript/examples/patch_account_pnl_configs.js
'use strict';

const { DNSEClient } = require('../dnse');

async function main() {
  const client = new DNSEClient({
    apiKey: 'replace-with-api-key',
    apiSecret: 'replace-with-api-secret',
    baseUrl: 'https://openapi.dnse.com.vn',
  });
  const payload = {
    takeProfit: {
      enabled: true, strategy: 'PNL_RATE', rate: 0.4, orderMethod: 'FASTEST', orderDeltaPrice: 2,
    },
    stopLoss: {
      enabled: true, strategy: 'DELTA_PRICE', rate: -0.1, orderMethod: 'DELTA_PRICE', orderDeltaPrice: 10.5, trailingEnabled: true,
    },
  };
  const { status, body } = await client.patchAccountPnlConfigs(
    '0001000115', 'DERIVATIVE', payload, 'replace-with-trading-token', { dryRun: false },
  );
  console.log(status, body);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: python/trading-api/patch_account_pnl_configs.py
#!/usr/bin/env python3
import os
import sys

sys.path.append(os.path.dirname(os.path.dirname(__file__)))

from dnse import DNSEClient


def main():
    client = DNSEClient(
        api_key="replace-with-api-key",
        api_secret="replace-with-api-secret",
        base_url="https://openapi.dnse.com.vn",
    )
    payload = {
        "takeProfit": {
            "enabled": True,
            "strategy": "PNL_RATE",
            "rate": 0.4,
            "orderMethod": "FASTEST",
            "orderDeltaPrice": 2.0,
        },
        "stopLoss": {
            "enabled": True,
            "strategy": "DELTA_PRICE",
            "rate": -0.1,
            "orderMethod": "DELTA_PRICE",
            "orderDeltaPrice": 10.5,
            "trailingEnabled": True,
        },
    }
    status, body = client.patch_account_pnl_configs(
        account_no="replace-with-account-no",
        market_type="DERIVATIVE",
        payload=payload,
        trading_token="replace-with-trading-token",
        dry_run=False,
    )
    print(status, body)


if __name__ == "__main__":
    main()
